src/main/scala/is/hail/nativecode/NativeCode.java

```diff
@@ -0,0 +1,40 @@
+package is.hail.nativecode;
+
+import java.util.*;
+import java.net.URL;
+import com.sun.jna.*;
+
+
+class NativeCode {
+  static {
+    System.err.println("NativeCode init ...");
+    //
+    // Access the library by the JNA which can find the
+    // library bundled into a JAR file, but doesn't use
+    // RTLD_GLOBAL to make symbols visible to everything.
+    //
+    NativeLibrary lib = NativeLibrary.getInstance("hail");
+    String libName = lib.getFile().toString();
+    System.err.println(libName);
+    lib.dispose();
+    //
+    // Load the library again with RTLD_GLOBAL
+    //
+    System.load(libName);
+    System.err.println("NativeCode init done");
```

Comment:
This makes me think we're not using JNA/JNI correctly.

Do we have two separate library files that are being loaded and cannot see one another's symbols?

I also see that we're not calling `Native.register` anywhere. My understanding of how to use JNA was to `Native.register("foo")` to define `native` methods using symbols from `libfoo`